docs/azure/sdk/snippets/dependency-injection/HostBuilder/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿using HostBuilder;
+#region snippet_HostBuilder
+using Azure.Identity;
+using Microsoft.Extensions.Azure;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Identity.Client.Platforms.Features.DesktopOs.Kerberos;
+
+IHost host = Host.CreateDefaultBuilder(args)
+    .ConfigureServices(services =>
+    {
+        services.AddHostedService<Worker>();
+        services.AddAzureClients(async clientBuilder =>
```

Comment:
This can't be async.   That forces a certain structuring.   I'd recommend adopting the pattern from the [Service Bus README](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/README.md#registering-with-aspnet-core-dependency-injection) here.   

IIn this case, you're hitting a problem because `ConfigureService` and `AddAzureClients` are `Action<T>` delegates - so you can't make them async safely.  

Hoisting the admin call out and stashing the queue names in a temporary list should work:

```csharp
// Query the available queues for the Service Bus namespace.
var adminClient = new ServiceBusAdministrationClient("<< SERVICE BUS CONNECTION STRING >>");
var queueNames = new List<string>();

// Because the result is async, they need to be captured to a standard list to avoid async
// calls when registering.  Failure to do so results in an error with the services collection.
await foreach (var queue in adminClient.GetQueuesAsync())
{
    queueNames.Add(queue.Name);
}

IHost host = Host.CreateDefaultBuilder(args)
    .ConfigureServices(services =>
    {
        services.AddHostedService<Worker>();
        services.AddAzureClients(clientBuilder =>
        {
            // Register clients for each service
            clientBuilder.AddSecretClient(new Uri("<key_vault_url>"));
            clientBuilder.AddBlobServiceClient(new Uri("<storage_url>"));
            clientBuilder.AddServiceBusClient("<NAMESPACE-NAME>.servicebus.windows.net");
            clientBuilder.UseCredential(new DefaultAzureCredential());

            foreach (var queueName in queueNames)
            {
                clientBuilder.AddClient<ServiceBusSender, ServiceBusClientOptions>((_, _, provider) =>
                    provider
                        .GetService<ServiceBusClient>()
                        .CreateSender(queueName)
                )
                .WithName(queueName);
            }
        });
    })
    .Build();

await host.RunAsync();
```